### PR TITLE
quosures in teal and deprecate old show r code approach

### LINKED ISSUE #640: Deprecate get_rcode_srv
Will be replaced by [more generic module](https://github.com/insightsengineering/teal.widgets/issues/30) - not dependent on chunks, datasets etc.

Blocked directly by:
- https://github.com/insightsengineering/teal.modules.general/issues/388
- https://github.com/insightsengineering/teal.modules.clinical/issues/479
- https://github.com/insightsengineering/teal.goshawk/issues/146
- https://github.com/insightsengineering/teal.osprey/issues/141

### PR DESCRIPTION
**Pull request by nikolas-burkoff**
Closes https://github.com/insightsengineering/teal/issues/682 and also closes #640 

- The only module I'm not sure what to do with is `filter_calls_module`

### CODE REVIEW

File: R/module_nested_tabs.R [hunk 1]
```diff
@@ -258,3 +227,38 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, reporter) {
...
+#' @return list of reactive datasets with following attributes:
+#' - `code` (`character`) containing datasets reproducible code.
+#' @keywords internal
+#' - `join_keys` (`JoinKeys`) containing relationships between datasets.
+.datasets_to_data <- function(module, datasets) {
+  datanames <- if (identical("all", module$filter)) {
+    datasets$datanames()
+    } else {
+    datasets$get_filterable_datanames(module$filter) # get_filterable_datanames adds parents if present
+    }
```
Comment (nikolas-burkoff) [inline R/module_nested_tabs.R:247]: Is there a reason we can't call `datasets$handle_active_datanames(modules$filter)` here?

Comment (nikolas-burkoff) [reply R/module_nested_tabs.R:247]: oh yeah I see we have the line `intersect(self$datanames(), datanames)` which means we can't...

**Comment by gogonzo** [comment]
@nikolas-burkoff let's wait with filter_calls_module until we resolve reactivity of attr(data, "code")

File: R/module_nested_tabs.R [hunk 2]
```diff
@@ -258,3 +227,38 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, reporter) {
...
+#' Converts `FilteredData` object to list of data containing datasets needed for specific module.
+#' Please note that if module needs dataset which has a parent, then parent will be also returned.
+#'
+#' @param module (`teal_module`) module where needed filters are taken from
+#' @param datasets (`FilteredData`) object where needed data are taken from
+#' @return list of reactive datasets with following attributes:
+#' - `code` (`character`) containing datasets reproducible code.
+#' @keywords internal
+#' - `join_keys` (`JoinKeys`) containing relationships between datasets.
+.datasets_to_data <- function(module, datasets) {
```
Comment (nikolas-burkoff) [inline R/module_nested_tabs.R:242]: I'll add a test for this one

File: R/module_nested_tabs.R [hunk 3]
```diff
@@ -258,3 +227,38 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, reporter) {
...
+      reactive(datasets$get_data(x, filtered = TRUE))
+    }
+  )
+
+  # code from previous stages
+  attr(data, "code") <- get_datasets_code(datanames, datasets)
+
+  # join_keys
+  attr(data, "join_keys") <- datasets$get_join_keys()
+  data
```
[the remark this answers is not in the document]
Comment (gogonzo) [reply R/module_nested_tabs.R:263]: In my opinion is better to wait with such decisions

File: R/module_nested_tabs.R (hunk 2 above)
Comment (nikolas-burkoff) [reply R/module_nested_tabs.R:242]: done

**Review by gogonzo — APPROVED** [review]
👍

**Pull request merged**